### PR TITLE
Even More Test Coverage

### PR DESCRIPTION
**Pull request by vkbo**
This PR improves the coverage of:

* Edit, Insert and Format menus. These tests have been modified to use the menu entries instead of jumping into the middle of the pipeline to post document actions directly. The coverage has also been improved.
* The auto-replace coverage has been extended, and many invalid actions are also checked in various parts of the document editor.
* Added test for the Search/Replace feature.
* Covered a lot of missing lines in document viewer, build tool, project tree, project class, and themes and icons classes.

This PR also fixes a number of corner case bugs that may or may not be reachable by a regular user, but was encountered by targeted tests.

### CODE REVIEW

File: nw/core/index.py [hunk 1]
```diff
@@ -356,7 +356,7 @@ def scanText(self, tHandle, theText):
 
         # Count words for remaining text after last heading
         if nTitle > 0:
-            lastText = "\n".join(theLines[nTitle-1:nLine-1])
+            lastText = "\n".join(theLines[nTitle-1:])
```
Comment (vkbo) [inline nw/core/index.py:359]: This one caused the section word count of the last line of a file to be missing.
